adaptive_navigation/lib/scaffold.dart

```diff
@@ -52,105 +46,132 @@ class AdaptiveNavigationScaffold extends StatefulWidget {
-      );
-    }
-
-    // Show a navigation rail
-    if (_isMediumScreen(context)) {
-      return Scaffold(
-        appBar: AppBar(
-          title: widget.title,
-        ),
-        body: Row(
+        );
+      case NavigationType.drawer:
+        // Show a Row containing a Drawer and Scaffold.
+        return Row(
           children: [
-            NavigationRail(
-              leading: widget.floatingActionButton,
-              destinations: [
-                ...widget.destinations.map(
-                  (d) => NavigationRailDestination(
-                    icon: Icon(d.icon),
-                    label: Text(d.title),
+            Drawer(
+              child: Column(
+                children: [
+                  DrawerHeader(
+                    child: Center(
+                      child: widget.title,
+                    ),
                   ),
-                ),
-              ],
-              selectedIndex: widget.currentIndex,
-              onDestinationSelected: widget.onNavigationIndexChange ?? (_) {},
+                  for (final destination in widget.destinations)
+                    ListTile(
+                      leading: Icon(destination.icon),
+                      title: Text(destination.title),
+                      selected: widget.destinations.indexOf(destination) ==
+                          widget.currentIndex,
+                      onTap: () => _destinationTapped(destination),
+                    ),
+                ],
+              ),
             ),
             VerticalDivider(
               width: 1,
               thickness: 1,
               color: Colors.grey[300],
             ),
             Expanded(
-              child: widget.body,
+              child: Scaffold(
+                appBar: AppBar(),
+                body: widget.body,
+                floatingActionButton: widget.floatingActionButton,
+              ),
             ),
           ],
-        ),
-      );
+        );
     }
-
-    // Show a bottom app bar
-    return Scaffold(
-      body: widget.body,
-      appBar: AppBar(title: widget.title),
-      bottomNavigationBar: BottomNavigationBar(
-        items: [
-          ...widget.destinations.map(
-            (d) => BottomNavigationBarItem(
-              icon: Icon(d.icon),
-              title: Text(d.title),
-            ),
-          ),
-        ],
-        currentIndex: widget.currentIndex,
-        onTap: widget.onNavigationIndexChange,
-      ),
-      floatingActionButton: widget.floatingActionButton,
-    );
   }
 
   void _destinationTapped(AdaptiveScaffoldDestination destination) {
-    var idx = widget.destinations.indexOf(destination);
-    if (idx != widget.currentIndex) {
-      widget.onNavigationIndexChange(idx);
+    final index = widget.destinations.indexOf(destination);
+    if (index != widget.currentIndex) {
+      widget.onNavigationIndexChange(index);
     }
   }
 }
+
+typedef NavigationType NavigationTypeResolver(BuildContext context);
```

Comment:
This typedef and the below enum might make sense to put at the top of the file since they are public?

Reply:
Move back up